File: sqldelight-sample/src/main/java/com/example/sqldelight/hockey/ui/PlayersActivity.java
package com.example.sqldelight.hockey.ui;

import android.app.Activity;
import android.arch.persistence.db.SupportSQLiteDatabase;
import android.content.Context;
import android.database.Cursor;
import android.os.Bundle;
import android.view.View;
import android.view.ViewGroup;
import android.widget.CursorAdapter;
import android.widget.ListView;
import butterknife.BindView;
import butterknife.ButterKnife;
import com.example.sqldelight.hockey.R;
import com.example.sqldelight.hockey.data.HockeyDb;
import com.example.sqldelight.hockey.data.Player;
import com.squareup.sqldelight.SqlDelightQuery;

public final class PlayersActivity extends Activity {
  public static final String TEAM_ID = "team_id";

  @BindView(R.id.list) ListView players;

  private Cursor playersCursor;

  @Override protected void onCreate(Bundle savedInstanceState) {
    super.onCreate(savedInstanceState);
    setContentView(R.layout.list);
    ButterKnife.bind(this);

    SupportSQLiteDatabase db = HockeyDb.getInstance(this).getReadableDatabase();
    long teamId = getIntent().getLongExtra(TEAM_ID, -1);
    if (teamId == -1) {
      SqlDelightQuery selectAllStatement = Player.FACTORY.select_all();
      playersCursor = db.query(selectAllStatement);
    } else {
      SqlDelightQuery playerForTeam = Player.FACTORY.for_team(teamId);
      playersCursor = db.query(playerForTeam);
    }
    players.setAdapter(new PlayersAdapter(this, playersCursor));
  }

  @Override protected void onDestroy() {
    super.onDestroy();
    playersCursor.close();
  }

  private static final class PlayersAdapter extends CursorAdapter {
    PlayersAdapter(Context context, Cursor c) {
      super(context, c, false);
    }

    @Override public View newView(Context context, Cursor cursor, ViewGroup parent) {
      return View.inflate(context, R.layout.player_row, null);
    }

    @Override public void bindView(View view, Context context, Cursor cursor) {
      Player.ForTeam playerForTeam = Player.FOR_TEAM_MAPPER.map(cursor);
      ((PlayerRow) view).populate(playerForTeam.player(), playerForTeam.team());
    }
  }
}

File: sqldelight-sample/src/main/java/com/example/sqldelight/hockey/ui/PlayersActivity.kt
package com.example.sqldelight.hockey.ui

import android.app.Activity
import android.os.Bundle
import android.support.v7.widget.LinearLayoutManager
import android.support.v7.widget.RecyclerView
import android.view.ViewGroup
import com.example.sqldelight.hockey.R
import com.example.sqldelight.hockey.data.ForTeam
import com.example.sqldelight.hockey.data.HockeyOpenHelper
import com.example.sqldelight.hockey.ui.PlayersActivity.PlayersAdapter.ViewHolder

class PlayersActivity : Activity() {
  override fun onCreate(savedInstanceState: Bundle?) {
    super.onCreate(savedInstanceState)
    setContentView(R.layout.list)

    val db = HockeyOpenHelper.getInstance(this).playerQueries
    val players = findViewById<RecyclerView>(R.id.list)
    players.layoutManager = LinearLayoutManager(this)
    players.adapter = PlayersAdapter(db.forTeam(intent.getLongExtra(TEAM_ID, -1)).executeAsList())
  }

  private inner class PlayersAdapter(
    private val data: List<ForTeam>
  ) : RecyclerView.Adapter<ViewHolder>() {
    override fun getItemCount() = data.size

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ViewHolder {
      return ViewHolder(layoutInflater.inflate(R.layout.player_row, parent, false) as PlayerRow)
    }

    override fun onBindViewHolder(holder: ViewHolder, position: Int) {
      holder.row.populate(data[position])
    }

    inner class ViewHolder(val row: PlayerRow): RecyclerView.ViewHolder(row)
  }

  companion object {
    val TEAM_ID = "team_id"
  }
}
